pennylane/qchem/tapering.py

```diff
@@ -756,32 +762,39 @@ def taper_operation(
 
     # build generator for the operation either internally or using the provided op_gen
     op_gen = _build_generator(operation, wire_order, op_gen=op_gen)
-    # check compatibility between the generator and the symmeteries
-    if np.all(
-        [
-            [
-                qml.is_commuting(op1, op2)
-                for op1, op2 in itertools.product(generator.ops, op_gen.ops)
-            ]
-            for generator in generators
-        ]
-    ) and not np.all(np.isclose(op_gen.coeffs, np.zeros_like(op_gen.coeffs), rtol=1e-8)):
-        gen_tapered = qml.taper(op_gen, generators, paulixops, paulix_sector)
-    else:
-        gen_tapered = qml.Hamiltonian([], [])
-    gen_tapered = qml.simplify(gen_tapered)
+
+    # Performing commutation check for pauli sentences
+    # TODO: replace when qml.is_commuting supports Pauli words and sentences
+    def _is_commuting(ps1, ps2):
+        commutator = ps1.commutator(ps2)
+        commutator.simplify()
+        return commutator == PauliSentence({})
+
+    # Obtain the tapered generator for the operation
+    with qml.QueuingManager.stop_recording():
+        # Get pauli rep for symmetery generators
+        ps_gen = list(map(lambda x: convert_to_opmath(x).pauli_rep, generators))
+
+        gen_tapered = PauliSentence({})
+        if all(_is_commuting(sym, op_gen) for sym in ps_gen) and not qml.math.allclose(
+            list(op_gen.values()), 0.0, rtol=1e-8
+        ):
+            gen_tapered = qml.taper(op_gen, generators, paulixops, paulix_sector)
+
+        gen_tapered = convert_to_opmath(gen_tapered).pauli_rep
```

Comment:
So `convert_to_opmath` can accept a `PauliSentence`? Do you know when that change was made, since I can't find that on master.

Reply:
```suggestion
            gen_tapered = qml.pauli.pauli_sentence(gen_tapered)
```

Reply:
Yep. It would return the `op` itself since [none](https://github.com/PennyLaneAI/pennylane/blob/bc2bb4abe8ebd3cbe299e2e937b2ded81c64e038/pennylane/operation.py#L3012) of the `if` conditions would match, i.e, the `op` is not a legacy operator.